File: app/src/main/java/com/tinashe/weather/model/TemperatureUnit.java
package com.tinashe.weather.model;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import androidx.annotation.StringDef;

@StringDef({TemperatureUnit.CELSIUS, TemperatureUnit.FAHRENHEIT})
@Retention(RetentionPolicy.RUNTIME)
public @interface TemperatureUnit {

    String CELSIUS = "celsius";

    String FAHRENHEIT = "fahrenheit";
}

File: app/src/main/java/com/tinashe/weather/data/model/TemperatureUnit.kt
package com.tinashe.weather.data.model

enum class TemperatureUnit(val value: String) {

    CELSIUS("celsius"),

    FAHRENHEIT("fahrenheit");

    companion object {
        private val map = values().associateBy(TemperatureUnit::value)

        fun fromValue(value: String) = map[value]
    }
}
